GitCommands/Settings/ConfigFileSettings.cs

```diff
@@ -70,16 +70,46 @@ public static ConfigFileSettings CreateGlobal(ConfigFileSettings? lowerPriority,
                 SettingLevel.SystemWide);
         }
 
-        public new string GetValue(string setting)
-        {
-            return GetString(setting, string.Empty);
-        }
+        public new string GetValue(string setting) => GetString(setting, string.Empty);
 
-        public IReadOnlyList<string> GetValues(string setting)
-        {
-            return SettingsCache.GetValues(setting);
+        /// <summary>
+        /// Get a git setting converted in an expected C# value type (bool, int, ...)
+        /// </summary>
+        /// <typeparam name="T">the expected type to convert the value to.</typeparam>
+        /// <param name="setting">the git setting key</param>
+        /// <returns>
+        /// null if the settings is not set
+        /// the value converted in the <typeparamref name="T" /> type otherwise.
+        /// </returns>
+        public T? GetValue<T>(string setting) where T : struct => ConvertValue<T>(GetValue(setting), setting);
+
+        private T? ConvertValue<T>(string value, string setting) where T : struct
+        {
+            // Handle case where setting is not set
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                return default;
```

Comment:
Not all git settings defaults match the default value of type. It should match the default setting value found in git config documentation.

Reply:
I don't think that's possible at this level. The caller has to provide the git default in this case.